doctests/dt-streams.js

```diff
@@ -0,0 +1,346 @@
+// EXAMPLE: stream_tutorial
+// HIDE_START
+import assert from 'assert';
+import {
+  createClient
+} from 'redis';
+
+const client = await createClient();
+await client.connect();
+// HIDE_END
+// REMOVE_START
+await client.flushDb();
+// REMOVE_END
+
+// STEP_START xAdd
+const res1 = await client.xAdd(
+  'race:france', '*', {
+    'rider': 'Castilla',
+    'speed': '30.2',
+    'position': '1',
+    'location_id': '1'
+  }
+);
+console.log(res1); // >>> 1700073067968-0 N.B. actual values will differ from these examples
+
+const res2 = await client.xAdd(
+  'race:france', '*', {
+    'rider': 'Norem',
+    'speed': '28.8',
+    'position': '3',
+    'location_id': '1'
+  },
+);
+console.log(res2); // >>> 1692629594113-0
+
+const res3 = await client.xAdd(
+  'race:france', '*', {
+    'rider': 'Prickett',
+    'speed': '29.7',
+    'position': '2',
+    'location_id': '1'
+  },
+);
+console.log(res3); // >>> 1692629613374-0
+// STEP_END
+
+// REMOVE_START
+assert.equal(await client.xLen('race:france'), 3);
+// REMOVE_END
+
+// STEP_START xRange
+const res4 = await client.xRange('race:france', '1691765278160-0', '+', 2);
```

Comment:
I feel a bit skeptical about these.  How can ranges work for streams when added with a '*' style IDs that would be based on timestamp work consistently (throughout this example).  Maybe I'm missing something?

Reply:
I don't know the streams data type well enough to comment intelligently, but the same examples are [present for the other 3 client libraries](https://redis.io/docs/latest/develop/data-types/streams/#querying-by-range-xrange-and-xrevrange). The person who created the initial CLI examples is long gone from Redis.

Reply:
ok.  I'll investigate a bit more, just feels weird to me, as I said, it might just be my lack of experience with them.